platform/logging/LoggingConfig.ts

```diff
@@ -0,0 +1,123 @@
+import * as schema from '../lib/schema';
+import { typeOfSchema } from '../types';
+import { LogLevel } from './LogLevel';
+import { BaseAppenderConfig, BaseAppenderConfigSchema } from './appenders/base/BaseAppenderConfig';
+import { ConsoleAppenderConfig, ConsoleAppenderConfigSchema } from './appenders/console/ConsoleAppenderConfig';
+import { FileAppenderConfig, FileAppenderConfigSchema } from './appenders/file/FileAppenderConfig';
+import { RollingFileAppenderConfig, RollingFileAppenderConfigSchema } from './appenders/rolling-file/RollingFileAppenderConfig';
+
+const CONTEXT_SEPARATOR = '::';
+const ROOT_CONTEXT_NAME = 'root';
+
+const KNOWN_APPENDERS = new Map<string, new (schema:  schema.TypeOf<typeof BaseAppenderConfigSchema>) => BaseAppenderConfig>([
+  ['console', ConsoleAppenderConfig],
+  ['file', FileAppenderConfig],
+  ['rolling-file', RollingFileAppenderConfig]
+]);
+
+const createLoggerSchema = () => {
+  return schema.object({
```

Comment:
Ok, will change, I was mostly concerned that without `schema` namespace `object` and `string` look like reserved `TS` words.